pandas/tests/frame/indexing/test_column_vs_index_tz.py

```diff
@@ -0,0 +1,57 @@
+import pandas as pd
+import pandas._testing as tm
+
+
+class TestColumnvsIndexTZEquality:
+    # https://github.com/pandas-dev/pandas/issues/29463
+    def check_for_various_tz(self, tz):
+        df = pd.DataFrame(
+            {
+                "val": range(10),
+                "time": pd.date_range(start="2019-01-01", freq="1d", periods=10, tz=tz),
+            }
+        )
+        df_query = df.query('"2019-01-03 00:00:00+00" < time')
+        l1 = pd.DataFrame(list(df_query["time"]))
+
+        # # This was earlier raising an exception.
+        index_query = df.set_index("time").query('"2019-01-03 00:00:00+00" < time')
+        l2 = pd.DataFrame(list(index_query.index))
+        tm.assert_frame_equal(l1, l2)
+
+    def test_check_column_vs_index_tz_query(self):
+        tz_list = [
+            "Africa/Abidjan",
+            "Africa/Douala",
+            "Africa/Mbabane",
+            "America/Argentina/Catamarca",
+            "America/Belize",
+            "America/Curacao",
+            "America/Guatemala",
+            "America/Kentucky/Louisville",
+            "America/Mexico_City",
+            "America/Port-au-Prince",
+            "America/Sitka",
+            "Antarctica/Casey",
+            "Asia/Ashkhabad",
+            "Asia/Dubai",
+            "Asia/Khandyga",
+            "Asia/Qatar",
+            "Asia/Tomsk",
+            "Atlantic/Reykjavik",
+            "Australia/Queensland",
+            "Canada/Yukon",
+            "Etc/GMT+7",
+            "Etc/UCT",
+            "Europe/Guernsey",
+            "Europe/Paris",
+            "Europe/Vienna",
+            "Indian/Cocos",
+            "NZ",
+            "Pacific/Honolulu",
+            "Pacific/Samoa",
+            "US/Eastern",
```

Comment:
Thanks @vipulrai91 

There's a fixture you can use here, `tz_aware_fixture` (feel free to ask for help if it's unclear how to use it)

Reply:
@MarcoGorelli thank you for the feedback.

As far as I understand the change has to be.        

```
def test_check_column_vs_index_tz_query(self, tz_aware_fixture):
    tz_list = tz_aware_fixture
```
Got this snippet from test_datetime.py , but what actually is tz_aware_fixture?
Also , How Do I run the test locally, I tried going through the docs , but will need some help here.

Thanks

Reply:
`tz = tz_aware_fixture` should be enough, then pytest will loop through the different timezones defined in `tz_aware_fixture`

> what actually is tz_aware_fixture?
It's a [fixture](https://docs.pytest.org/en/latest/fixture.html) which you can use to test different timezones.

> How Do I run the test locally

See [using pytest](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#using-pytest). So here,
```bash
$ pytest pandas/tests/frame/indexing/test_column_vs_index_tz.py
```
should be enough

Reply:
Thank you for guiding.

tz_aware_fixture is a combination of TimeZones and other object types , also even 
it throws pytz.exceptions.UnknownTimeZone error.



```
test_column_vs_index_tz.py:58: TypeError
============================================================ short test summary info =============================================================
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query['UTC'] - pytz.exceptions.UnknownTimeZoneErr...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query['US/Eastern'] - pytz.exceptions.UnknownTime...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query['Asia/Tokyo'] - pytz.exceptions.UnknownTime...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query['dateutil/US/Pacific'] - pytz.exceptions.Un...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query['dateutil/Asia/Singapore'] - pytz.exception...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[tzutc()] - TypeError: 'tzutc' object is not...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[tzlocal()] - TypeError: 'tzlocal' object is...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[pytz.FixedOffset(300)] - TypeError: '_Fixed...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[<UTC>] - TypeError: 'UTC' object is not ite...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[pytz.FixedOffset(-300)] - TypeError: '_Fixe...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[datetime.timezone.utc] - TypeError: 'dateti...
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[datetime.timezone(datetime.timedelta(seconds=3600))]
FAILED test_column_vs_index_tz.py::TestColumnvsIndexTZEquality::test_check_column_vs_index_tz_query[datetime.timezone(datetime.timedelta(days=-1, seconds=82800), 'foo')]
```